lib/github.dart

```diff
@@ -0,0 +1,1029 @@
+    }
+  }
+
+  void _updateGist() {
+    final String token = _githubAuthController.githubOAuthAccessToken;
+    if (token.isNotEmpty) {
+      final Gist clonedGist = _playground.mutableGist.createGist();
+      gistLoader.updateGist(clonedGist, token).then((String updatedGistId) {
+        setUnsavedLocalEdits();
+        _playground.showSnackbar('Gist successfully updated');
+
+        // Update the backing gist because it is now in github.
+        _playground.mutableGist.setBackingGist(clonedGist);
+
+        // Now update our menus to reflect new gist (description could have changed).
+        _githubAuthController.updateUsersGistAndStarredGistsList();
+      });
+    } else {
+      _playground.showSnackbar(
+          'Must be authenticated with GitHub in order to fork gist');
+    }
+  }
+
+  void _forkGist() {
+    final String token = _githubAuthController.githubOAuthAccessToken;
+    final bool unsavedLocalEdits = _playground.mutableGist.dirty;
+    if (token.isNotEmpty) {
+      gistLoader
+          .forkGist(
+              _playground.mutableGist.createGist(), unsavedLocalEdits, token)
+          .then((String forkedGistId) {
+        if (forkedGistId == 'GIST_ALREADY_FORK') {
+          _playground.showSnackbar('Failed to fork gist - already a fork');
+          return;
+        } else if (forkedGistId == 'GIST_NOT_FOUND') {
+          _playground.showSnackbar('Failed to fork gist - gist not found');
+          return;
+        }
+
+        setUnsavedLocalEdits();
+
+        _playground.showSnackbar(unsavedLocalEdits
+            ? 'Gist successfully forked and updated with local edits'
+            : 'Gist successfully forked'); // This wont have time to show KLUDGE
+
+        _reloadPageWithNewGistId(forkedGistId);
+
+        // Now update our menus to reflect new gist.
+        _githubAuthController.updateUsersGistAndStarredGistsList();
+      });
+    } else {
+      _playground.showSnackbar(
+          'Must be authenticated with GitHub in order to fork gist');
+    }
+  }
+
+  void _reloadPageWithNewGistId(String gistId) {
+    var url = Uri.parse(window.location.toString());
+    final params = Map<String, String?>.from(url.queryParameters);
+    params['id'] = gistId;
+    url = url.replace(queryParameters: params);
+    window.location.href = url.toString();
+  }
+
+  void _setMenuItemState(LIElement menuitem, bool enabled) {
+    if (enabled) {
+      menuitem.classes.remove('mdc-list-item--disabled');
+    } else {
+      menuitem.classes.add('mdc-list-item--disabled');
+    }
+  }
+
+  String _truncateWithEllipsis(String text, int maxlength,
+      {String ellipsis = '...'}) {
+    return (text.length < maxlength)
+        ? text
+        : text.replaceRange(maxlength, text.length, ellipsis);
+  }
+
+  void _myGistMenuHandler(Event e) {
+    final index = (e as CustomEvent).detail['index'] as int;
+    final List<Gist> mygists = _githubAuthController.myGistList;
```

Comment:
The type can be inferred:

```suggestion
    final mygists = _githubAuthController.myGistList;
```